app/viewmodels/RegistrationDetailsBuilder.scala

```diff
@@ -0,0 +1,178 @@
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels
+
+import connectors.ConfigConnector
+import controllers.registration.returns.{routes => returnsRoutes}
+import featureswitch.core.config.FeatureSwitching
+import models.{BankAccount, BankAccountDetails, BeingSetup, FlatRateScheme, NameChange, OverseasAccount, OverseasBankDetails}
+import models.api.returns.{Annual, Monthly, Quarterly, Returns}
+import models.api.{NETP, PartyType, VatScheme}
+import models.view.SummaryListRowUtils.{optSummaryListRowBoolean, optSummaryListRowSeq, optSummaryListRowString}
+import play.api.i18n.Messages
+import services.FlatRateService
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.{SummaryList, SummaryListRow}
+import uk.gov.hmrc.http.InternalServerException
+
+import java.time.format.DateTimeFormatter
+import javax.inject.{Inject, Singleton}
+
+// scalastyle:off
+@Singleton
+class RegistrationDetailsBuilder @Inject()(configConnector: ConfigConnector,
+                                           flatRateService: FlatRateService) extends FeatureSwitching {
+
+  val presentationFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("d MMMM y")
+  val sectionId = "cya.registrationDetails"
+
+  def build(vatScheme: VatScheme)(implicit messages: Messages): SummaryList = {
+    val partyType = vatScheme.eligibilitySubmissionData.map(_.partyType).getOrElse(throw new InternalServerException("Eligibility"))
+    val returns = vatScheme.returns.getOrElse(throw new InternalServerException("[RegistrationDetailsBuilder] Returns"))
+
+    SummaryList(
+      List(
+        startDate(returns, partyType),
+        accountingPeriod(returns),
+      ).flatten ++
+        bankAccountSection(vatScheme, partyType) ++
+        flatRateSchemeSection(vatScheme, partyType)
+    )
+  }
+
+  private def startDate(returns: Returns, partyType: PartyType)(implicit messages: Messages): Option[SummaryListRow] =
+    optSummaryListRowString(
+      s"$sectionId.startDate",
+      returns.startDate match {
+        case Some(date) => Some(date.format(presentationFormatter))
+        case None if partyType.equals(NETP) => None
+        case None => Some(s"$sectionId.mandatoryStartDate")
+      },
+      Some(controllers.registration.returns.routes.VatRegStartDateResolverController.resolve.url)
+    )
+
+  private def accountingPeriod(returns: Returns)(implicit messages: Messages): Option[SummaryListRow] =
```

Comment:
When user selects Annual there's also a few extra AAS pages that all fall into this cya block